### PR TITLE
[deckhouse-controller] ModulePullOverride resource

### PR DESCRIPTION
**Pull request by yalosev**
## Description
Add new resource `ModulePullOverride`  to have ability to set and pull custom module images.

## Why do we need it, and what problem does it solve?
ModuleSource is very inflexible to operate with modules in developer mode. New resource makes module developing mode flexible

## What is the expected result?
Creating `ModulePullOverride` resource:
```
apiVersion: deckhouse.io/v1alpha1
kind: ModulePullOverride
metadata:
  name: echo
spec:
  imageTag: pr123
  scanInterval: 15s
  source: test
```

will pull docker image with tag `pr123` for module `echo` from source `test` every 15 seconds. If image digest have been changed, module will be reloaded.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: feature
summary: Add new resource ModulePullOverride to pull and apply modules in the development mode.
```

### CODE REVIEW

**Review by nabokihms — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

File: deckhouse-controller/crds/doc-ru-module-release.yaml [hunk 1]
```diff
@@ -14,14 +13,12 @@ spec:
...
                 weight:
-                  type: number
                   description: _Вес_ (приоритет) модуля.
             status:
               properties:
                 phase:
                   description: Текущий статус релиза.
                 message:
-                  type: string
                   description: Детальное сообщение об ошибки или статусе релиза.
```
Comment (z9r5) [inline deckhouse-controller/crds/doc-ru-module-release.yaml:22]:
```suggestion
                  description: Детальное сообщение об ошибке или статусе релиза.
```

File: deckhouse-controller/crds/doc-ru-module-source.yaml [hunk 2]
```diff
@@ -43,8 +43,8 @@ spec:
                       policy:
                         type: string
                         description: Имя политики.
-                  type: array
-                  description: Список доступных для установки модулей в данном репозитории.
+                      overrided:
+                        description: Показывает, есть ли ModulePullOverride ресурс для данного модуля.
```
Comment (z9r5) [inline deckhouse-controller/crds/doc-ru-module-source.yaml:47]:
```suggestion
                        description: Показывает, есть ли ресурс ModulePullOverride для данного модуля.
```

File: deckhouse-controller/crds/module-pull-override.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+              required:
+                - source
+                - imageTag
+              properties:
+                source:
+                  type: string
+                  description: "Reference to the ModuleSource with the module."
+                imageTag:
+                  type: string
+                  description: "Docker image tag for module, which will be pulled."
```
Comment (z9r5) [inline deckhouse-controller/crds/module-pull-override.yaml:42]:
```suggestion
                  description: "Module container image tag, which will be pulled."
```

File: deckhouse-controller/crds/module-pull-override.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                scanInterval:
+                  type: string
+                  description: "Scan interval for checking the image digest. If digest is changed, module will be pulled."
```
Comment (z9r5) [inline deckhouse-controller/crds/module-pull-override.yaml:45]:
```suggestion
                  description: "Scan interval for checking the image digest. If the digest changes, the module is updated."
```

File: deckhouse-controller/crds/module-pull-override.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                  default: 15s
+            status:
+              type: object
+              properties:
+                message:
+                  type: string
+                  description: "Details of the resource status."
+                updatedAt:
+                  type: string
+                  description: "When the module was last downloaded."
```
Comment (z9r5) [inline deckhouse-controller/crds/module-pull-override.yaml:55]:
```suggestion
                  description: "When the module was last updated."
```

File: deckhouse-controller/crds/module-pull-override.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                  type: string
+                  description: "Digest of the module image."
+      subresources:
+        status: {}
+      additionalPrinterColumns:
+        - name: "Updated"
+          type: date
+          jsonPath: .status.updatedAt
+          format: date-time
+          description: 'When the module was last downloaded.'
```
Comment (z9r5) [inline deckhouse-controller/crds/module-pull-override.yaml:66]:
```suggestion
          description: 'When the module was last updated.'
```

**Pull request merged**